pulpcore/pulpcore/app/response.py

```diff
@@ -31,7 +31,8 @@ def __init__(self, task_results, request):
         """
         tasks = []
         for result in task_results:
-            task = {"_href": reverse('tasks-detail', args=[result.task_id], request=request),
+            task = {"_href": reverse('tasks-core-updates-detail', args=[result.task_id],
+                                     request=request),
```

Comment:
ODO: Can this be generalized? 

something like:

    task = CoreTask.objects.get(id=result.task_id).cast()
    
Except I don't think the model object knows about its serializer. Could we find some way to use a serializer to have DetailRelatedField(many=True) that would cast() to render with the detail serializer? Would replace "OperationPostponedResponse"

Maybe here? a new serializer that is used here, and in the AsyncRemoveMixin https://github.com/pulp/pulp/pull/3394/files#diff-ae7f375b1b9eaee42aecdf250679cd6eR232

Reply:
yeah, esp. in case this was an async result of some plug-in specific thing, returning this hard-coded `tasks-core-updates-detail` reverse might not be cool. 

I think part of the issue is that we change this logic to be an object-like endpoint instead a verb; posting to a serializer to create an object, we might want to avoid returning a list of them, it feels odd.

If instead there was a `WaitingTask` that had a custom, read-only serializer, a read only `parent` field  and a `RunningTask` had a read-only `children` field in addition, we might  avoid having to have a custom response handling logic (here) and make the tasks appear what they actually are, a lazily-unspooling, directed acyclic graph.

See also my off-topic comment below

Reply:
The OperationPostponed will only ever be used for asynchronous updates/deletes. The reason for this is that these tasks are created by the ViewSets for the objects that are being updated/deleted (repos, importers, publishers).

Other tasks are created/read/listed at their own endpoints and handled by their own viewsets.